cmd/snap-confine/mount-support.c

```diff
@@ -300,7 +304,30 @@ static void sc_bootstrap_mount_namespace(const struct sc_mount_config *config)
 		}
 		sc_must_snprintf(dst, sizeof dst, "%s/%s", scratch_dir,
 				 mnt->path);
-		sc_do_mount(mnt->path, dst, NULL, MS_REC | MS_BIND, NULL);
+		// NOTE: we are not using sc_do_mount because we want non-fatal errors.
+		if (mount(mnt->path, dst, NULL, MS_REC | MS_BIND, NULL) < 0) {
+			if (errno == ENOENT && mnt->is_optional) {
+				// When an optional mount entry fails with ENOENT then just
+				// carry on. This represents a missing mount point in the base
+				// snap or missing mount source on the host.
+				continue;
+			}
+			char mount_cmd_buf[10000] = { 0 };
```

Comment:
A comment here that says: `// Non-optional mount failed. Create a useful error message` would be nice.